### PR TITLE
Receipts are sorted by date (Newest First) - Issue #72

### PR DESCRIPTION
**Pull request by ezhil56x**
Now the receipts will be sorted based on Date (Newest First) and will be displayed - Issue #72

### CODE REVIEW

**Comment by therealsujitk** [comment]
@ezhil56x thanks for your contribution, I'll review an merge this as soon as possible.

**Comment by ezhil56x** [comment]
Your welcome!

**Comment by therealsujitk** [comment]
If you're busy, I can make these changes myself when I'm free.

**Comment by ezhil56x** [comment]
@therealsujitk I'm not sure what should I do?

**Comment by therealsujitk** [comment]
> @therealsujitk I'm not sure what should I do?

I've written down all the changes that need to be made in my review.

image

File: app/src/main/java/tk/therealsuji/vtopchennai/fragments/RecyclerViewFragment.java [hunk 1]
```diff
@@ -61,6 +64,21 @@ public void onSuccess(@NonNull List<Receipt> receipts) {
...
+                                @SuppressLint("SimpleDateFormat") SimpleDateFormat format = new SimpleDateFormat("dd-MMM-yyyy");
+                                try {
+                                    return format.parse(r2.getDate()).compareTo(format.parse(r1.getDate()));
+                                } catch (Exception e) {
+                                    e.printStackTrace();
+                                }
+                                return 0;
+                            }
+                        });
+// Code Ends Here
```
Comment (therealsujitk) [inline app/src/main/java/tk/therealsuji/vtopchennai/fragments/RecyclerViewFragment.java:80]: A few changes here,

- The comments are unnecessary, it's understood what's going on here.
- Avoid suppressing warnings (`@SuppressLint`). The warning appears due to the missing locale.
  ```java
  SimpleDateFormat format = new SimpleDateFormat("dd-MMM-yyyy", Locale.ENGLISH);
  ```
- Printing the stack trace is unnecessary, ignore the exception instead.
  ```diff
  - } catch (Exception e) {
  + } catch (Exception ignored) {
  -     e.printStackTrace();
    }
  ```

File: app/src/main/java/tk/therealsuji/vtopchennai/models/Receipt.java [hunk 2]
```diff
@@ -14,4 +14,9 @@ public class Receipt {
 
     @ColumnInfo(name = "date")
     public String date;
+
+    public String getDate() {
+        return date;
+    }
+
```
Comment (therealsujitk) [inline app/src/main/java/tk/therealsuji/vtopchennai/models/Receipt.java:21]: Can you revert these changes. The `date` property is public, so accessing it via a method is unnecessary. `r1.date` should be sufficient.

**Comment by ezhil56x** [comment]
I'll make these changes

**Comment by ezhil56x** [comment]
@therealsujitk All the changes you have suggested are done

**Review by therealsujitk — APPROVED** [review]
